homeassistant/components/light/zwave.py

```diff
@@ -0,0 +1,132 @@
+"""
+homeassistant.components.light.zwave
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+"""
+# pylint: disable=import-error
+from openzwave.network import ZWaveNetwork
+from pydispatch import dispatcher
+
+import homeassistant.components.zwave as zwave
+
+from homeassistant.const import STATE_ON, STATE_OFF
+from homeassistant.components.light import (Light, ATTR_BRIGHTNESS)
+from time import sleep
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Find and add zwave lights. """
+    if discovery_info is None:
+        return
+
+    node = zwave.NETWORK.nodes[discovery_info[zwave.ATTR_NODE_ID]]
+    value = node.values[discovery_info[zwave.ATTR_VALUE_ID]]
+
+    if value.command_class != zwave.COMMAND_CLASS_SWITCH_MULTILEVEL:
+        return
+    if value.type != zwave.TYPE_BYTE:
+        return
+    if value.genre != zwave.GENRE_USER:
+        return
+
+    value.set_change_verified(False)
+    add_devices([ZwaveDimmer(value)])
+
+
+def brightness_state(value):
+    """ Returns the brightness and state according to the current
+    data of given value. """
+    if value.data > 0:
+        return (value.data / 99) * 255, STATE_ON
+    else:
+        return 255, STATE_OFF
+
+
+class ZwaveDimmer(Light):
+    """ Provides a zwave dimmer. """
+    # pylint: disable=too-many-arguments
+    def __init__(self, value):
+        self._value = value
+        self._node = value.node
+
+        self._brightness, self._state = brightness_state(value)
+
+        # Used for value change event handling
+        self._refreshing = False
+        self._expect = None
+
+        dispatcher.connect(
+            self._value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+
+    def _value_changed(self, value):
+        """ Called when a value has changed on the network. """
+        if self._value.value_id == value.value_id:
```

Comment:
I see a lot of if with no else statements. Consider using guard  clause pattern to avoid nested ifs :

_typed from my phone so no real syntax_

If value not equal :
  Return 

If not refreshing :
  Refreshing = true 
  Return